doc/source/reference/apis/metadata.md

```diff
@@ -22,7 +26,97 @@ this feature graduates. If you have any comments or suggestion please open the i
 
 ## Model Metadata (incubating)
 
-With Seldon you can easily define metadata for your models that is compatible with [kfserving dataplane proposal](https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md#model-metadata) specification:
+With Seldon you can easily add metadata to your models.
+You can add metadata using one of two formats:
+- `v1` format that closely correlate to the current structure of `SeldonMessage`
+- `v2` format that is future-proof and fully compatible with `kfserving` dataplane proposal
+
+As you will see in following sections difference is minimal and mostly relates to the input/output format.
+
+
+### V1 DataPlane
+
+In order to use `v1` metadata format you need to specify `apiVersion: v1`.
+
+#### Array input/output
+```YAML
+apiVersion: v1
+name: my-model-name
+versions: [ my-model-version-01 ]
+platform: seldon-custom
+inputs:
+  datatype: array
+  shape: [ 2, 2 ]
+outputs:
+  datatype: array
+  shape: [ 1 ]
+```
+This metadata would mean that following two inputs are valid for this model:
+```JSON
+{"data": {"names": ["input"], "ndarray": [[1, 2], [3, 4]]}}
+```
+and
+```JSON
+{"data": {"names": ["input"], "tensor": {"values": [1, 2, 3, 4], "shape": [2, 2]}}
+```
+
+#### jsonData input/output
+```YAML
+apiVersion: v1
+name: my-model-name
+versions: [ my-model-version-01 ]
+platform: seldon-custom
+inputs:
+  datatype: jsonData
```

Comment:
nit: should we just call the new datatype `json` instead of `jsonData`?

I see the `*Data` suffix as just a way to differentiate in the payload between `data`, `strData` and `jsonData`.

Reply:
Also I chose `jsonData` so it is the same as what you specify in the payload when you want to send it.

Reply:
But following that logic for `ndarray` and `tensor` (that I capture at single option `array`) I should use `data`.... Yeah, I don't mind removing `*Data` suffix.